### ISSUE TITLE
FR: Implement $dt->days, $dt->last_day

### ISSUE THREAD

**Issue by KES777** [houseabsolute/DateTime.pm#66]
I see this function useful when we want to get count of days in current month or DateTime object for last day in current month.

Please implement if you see them useful too

*FR: Feature request

**Comment by autarch** [comment]
See #60 for a discussion of pretty much exactly this feature.

**Comment by autarch** [comment]
There's a suggestion in that issue of adding methods for length of month, quarter, & year. That does seem reasonable to me.

**Comment by danstewart** [comment]
@autarch Happy to try implementing this. Is your preference still class methods?

**Comment by autarch** [comment]
@danstewart - I think @chansen convinced me that these should just be object methods.

I think the write naming scheme is:

* `$dt->length_of_{month,quarter,year}`
* `$dt->is_last_day_of_{month,quarter,year}`

Note that `is_last_day_of_month` already exists.

**Comment by KES777** [comment]
Would it be better if: `->month_length`, `->qarter_length`, `->year_length`?

this is shorter in compare with other versions

**Comment by autarch** [comment]
@KES777 - good point, that works for me.

**Comment by chansen** [comment]
@autarch wouldn't that make the method naming scheme inconsistent?

`is_last_day_of_{component} = day_of_{component} == {component}_length`

```perl
$dt->is_last_day_of_year    = $dt->day_of_year    == $dt->year_length;
$dt->is_last_day_of_quarter = $dt->day_of_quarter == $dt->quarter_length;
$dt->is_last_day_of_month   = $dt->day_of_month   == $dt->month_length;
```

Versus:

`is_last_day_of_{component} = day_of_{component} == length_of_{component}`

```perl
$dt->is_last_day_of_year    = $dt->day_of_year    == $dt->length_of_year;
$dt->is_last_day_of_quarter = $dt->day_of_quarter == $dt->length_of_quarter;
$dt->is_last_day_of_month   = $dt->day_of_month   == $dt->length_of_month;
```

**Comment by autarch** [comment]
I think we should generally prefer shorter names as long as they're clear. This trumps consistency. The other `X_of_Y` methods can't really be shortened without losing clarity.

**Comment by autarch** [comment]
This has been implemented since 1.45.
